app/screens/post_options/post_options.js

```diff
@@ -285,6 +309,12 @@ export default class PostOptions extends PureComponent {
         });
     };
 
+    handleToggleFollow = () => {
+        const {actions, currentTeamId, currentUserId, post} = this.props;
+        actions.setThreadFollow(currentUserId, currentTeamId, post.id, !post.is_following);
```

Comment:
careful with the currentTeamId, this should be the team that the post belongs to and only the currentTeamId if the post is in a DM/GM.

I guess you can actually fix an existing bug here with the `copy link` option ;)